### PR TITLE
Experimental ws for H3Grafserv

### PR DESCRIPTION
**Pull request by Dodobibi**
This PR add :
- bump to h3@1.8.1
- replace `getMethod` (deprecated in h3@1.8.1) with `event.method`
- `handleEvent`, `handleGraphiqlEvent` and `handleEventStreamEvent` helpers (for using with nuxt3)
- `addTo` to H3Grafserv (for using with h3 as standalone),
- add an experimental `H3Grafserv` (when importing via `server/h3/v1/experimental') for implements subsciption-ws for h3 (or nuxt3)

### CODE REVIEW

**Comment by Dodobibi** [comment]
Yes,  It was just for making the new methods clearly experimental.
New commit rebased.

**Comment by benjie** [comment]
Nice! From a quick scan this is looking really good (will do a better review later). Have you tested it well and are happy it works? Could you add a page to the documentation for it; it would be a `h3.md` file created here:

https://github.com/benjie/crystal/tree/main/grafast/website/grafserv/servers

Ideally you should document how to use it with a nuxt app ❤️

**Comment by Dodobibi** [comment]
> Have you tested it well and are happy it works?

Yes, for h3 as a standalone server and for nuxt too.

> Could you add a page to the documentation for it

Yes, I will try to draft this, but you will have to (or @jemgillam) review my English syntax...

> Ideally you should document how to use it with a nuxt app

Of course.

**Comment by benjie** [comment]
We tend to always edit the documentation whether the author is a native English speaker or not, so don't worry about that! Mostly I care about the instructions themselves, since I don't currently use Nuxt.

**Comment by peteromano** [comment]
Hi guys! I know you're busy this week benjie, any chance of merging this this week?

If not, I'll pull the branch and work off/test that for now.

@Dodobibi fyi I'm also working on nuxt 3 integration. Can collab of the discord channel @peteromano 🙂

**Comment by peteromano** [comment]
Yea it seemed to work yesterday, gonna spend some more time with it later this week, no rush.

File: grafast/grafserv/src/servers/h3/v1/index.ts [hunk 1]
```diff
@@ -76,7 +81,15 @@ export class H3Grafserv extends GrafservBase {
   constructor(config: GrafservConfig) {
     super(config);
   }
+
+  /**
+   * @deprecated use handleGraphqlEvent instead
```
Comment (benjie) [inline grafast/grafserv/src/servers/h3/v1/index.ts:86]:
```suggestion
   * @deprecated use handleGraphQLEvent instead
```

File: grafast/grafserv/src/servers/h3/v1/index.ts [hunk 2]
```diff
@@ -76,7 +81,15 @@ export class H3Grafserv extends GrafservBase {
...
+   */
   public async handleEvent(event: H3Event) {
+    return this.handleGraphqlEvent(event);
```
Comment (benjie) [inline grafast/grafserv/src/servers/h3/v1/index.ts:89]:
```suggestion
    return this.handleGraphQLEvent(event);
```

File: grafast/grafserv/src/servers/h3/v1/index.ts [hunk 3]
```diff
@@ -76,7 +81,15 @@ export class H3Grafserv extends GrafservBase {
...
+  }
+
+  public async handleGraphqlEvent(event: H3Event) {
```
Comment (benjie) [inline grafast/grafserv/src/servers/h3/v1/index.ts:92]:
```suggestion
  public async handleGraphQLEvent(event: H3Event) {
```

File: grafast/grafserv/src/servers/h3/v1/index.ts [hunk 4]
```diff
@@ -176,6 +211,90 @@ export class H3Grafserv extends GrafservBase {
...
+
+  public async addTo(app: App) {
+    const dynamicOptions = this.dynamicOptions;
+
+    const router = createRouter();
+    app.use(router);
+
+    router.use(
+      this.dynamicOptions.graphqlPath,
+      eventHandler((event) => this.handleGraphqlEvent(event)),
```
Comment (benjie) [inline grafast/grafserv/src/servers/h3/v1/index.ts:223]:
```suggestion
      eventHandler((event) => this.handleGraphQLEvent(event)),
```

**Review by benjie — CHANGES_REQUESTED** [review]
Looks good to me!

**Review by benjie — APPROVED** [review]

**Pull request merged**